src/Controller.php

```diff
@@ -51,27 +51,34 @@ public function __construct(
      */
     public static function init()
     {
-        $opts = getopt('', array('dir:','replace'));
+        list($composer, $mode, $output) = self::getParams();
         return Container::create()
-            ->addValue(
-                'SplFileInfo.file_name',
-                dirname(__DIR__).DIRECTORY_SEPARATOR.'templates'
-            )
-            ->addValue(
-                'PhpParser\Parser.options',
-                array('throwOnError'=>true)
-            )
-            ->addValue(
-                'De\Idrinth\TestGenerator\Interfaces\JsonFile.file',
-                (isset($opts['dir']) && $opts['dir'] ?
-                        rtrim($opts['dir'], DIRECTORY_SEPARATOR) :
-                        getcwd()
-                    ).DIRECTORY_SEPARATOR.'composer.json'
-            )
-            ->addValue('De\Idrinth\TestGenerator\Interfaces\ClassWriter.replace', isset($opts['replace']))
+            ->addValue('SplFileInfo.file_name', dirname(__DIR__).DIRECTORY_SEPARATOR.'templates')
+            ->addValue('PhpParser\Parser.options', array('throwOnError' => true))
+            ->addValue('De\Idrinth\TestGenerator\Interfaces\JsonFile.file', $composer)
+            ->addValue('De\Idrinth\TestGenerator\Interfaces\ClassWriter.mode', $mode)
+            ->addValue('De\Idrinth\TestGenerator\Interfaces\Composer.output', $output)
             ->get(__CLASS__);
     }
 
+    /**
+     * processes getopt
+     * @return array [$composer, $mode, $output]
+     */
+    private static function getParams()
+    {
+        $opts = getopt('', array('dir:','replace','output:','mode:'));
+        $dir = (isset($opts['dir']) && $opts['dir'] ? rtrim($opts['dir'], '\\/') : getcwd());
```

Comment:
i dont get it. $opts['dir'] is a boolean? if its true, youre going to trim a boolean. Or what is evaluated there?

Reply:
luckily this will be gone once we have the handler for cli-access, what it does for now is:
- has someone set the option
- does the option have a value